lawlibrary/views/court.py

```diff
@@ -0,0 +1,66 @@
+from django.shortcuts import get_object_or_404
+
+from africanlii.utils import lowercase_alphabet
+from peachjam.models import Author, Judgment
+from peachjam.views.generic_views import FilteredDocumentListView
+
+
+class BaseCourtDetailView(FilteredDocumentListView):
+    """Generic List View class for filtering a court's judgments."""
+
+    context_object_name = "documents"
+    paginate_by = 20
+    model = Judgment
+    template_name = "lawlibrary/court_detail.html"
+
+    def get_base_queryset(self):
+        if "year" in self.kwargs:
+            return self.model.objects.filter(
+                author=self.author, date__year=self.kwargs["year"]
+            )
+        return self.model.objects.filter(author=self.author)
+
+    def get_queryset(self):
+        self.author = get_object_or_404(Author, code=self.kwargs["code"])
+        return super().get_queryset()
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        years = list(
```

Comment:
I know we use this pattern elsewhere, but as our DB grows we'll land up loading 1000s of years into memory just to make them distinct. Much more efficient to get the db to do it:

```python
years = self.model.objects.filter(author=self.author).order_by().values_list("date__year", flat=True).distinct()
```

This is a pattern we can re-use elsewhere. The `order_by()` is important to turn off the default ordering, which would be doing the wrong thing.

Here's an example of what the SQL is, just to confirm:

```
>>> str(Judgment.objects.filter(author__name="foo").order_by().values_list("date__year", flat=True).distinct().query)
'SELECT DISTINCT EXTRACT(\'year\' FROM "peachjam_coredocument"."date") FROM "peachjam_judgment"
INNER JOIN "peachjam_author" ON ("peachjam_judgment"."author_id" = "peachjam_author"."id")
INNER JOIN "peachjam_coredocument" ON ("peachjam_judgment"."coredocument_ptr_id" = 
"peachjam_coredocument"."id") WHERE "peachjam_author"."name" = foo'
```